### PR TITLE
LOOP-1926: Override user language setting by forcing english

### PR DESCRIPTION
**Pull request by rickpasetto**
This uses a trick that forces setting the "AppleLanguages" UserDefault if FORCE_ENGLISH is set as a compiler flag.  This seemed the lightest-weight approach towards solving [LOOP-1926](https://tidepool.atlassian.net/browse/LOOP-1926).  Setting "AppleLanguages" back to `nil` seems to restore the user's preference (when tested in the simulator).

### CODE REVIEW

**Review by ps2 — APPROVED** [review]
Not a big fan that we're disabling translations, but I understand the reasoning about not enabling translations that were not reviewed by Tidepool.  

This method is using a non-documented approach to change the language. Adding an app-wide setting that ends up being used in all calls to NSLocalizedString would be more in-line with expected use of the API. But perhaps we can get by with this hack for now.

**Comment by rickpasetto** [comment]
@ps2 No doubt, this is a bit of a hack. But in my explorations, there's really no clean way to do this other than to replace all `NSLocalizedString` calls (and their ilk) with something of our making and then have that facility be the way to "force" a language.

**Comment by ps2** [comment]
You'd still use NSLocalizedString, but it would be through a helper (like those we use now) that provided bundle with a specific language path.

**Comment by rickpasetto** [comment]
@ps2 I've done some more investigation of this.  Here were some alternatives to this PR's solution that I considered:

1. Changing all calls to `NSLocalizedString`, `LocalizedString`, etc. to go through a centralized function that would "override" to English (en-US).  There were several problems with this approach: 
- We'd need to have multiple copies of this function, implemented in every Framework
- The implementations still would have to "poke" into the Bundle to find the `path` to the `en` (or `Base`) `lproj` directory for the right resource.  This seems to circumvent the `Locale` system, and seems error prone.
- it would not cleanly take care of localizing `Text()` views in SwiftUI

2. Having some kind of script or "package" step that would only pluck the string files for the language(s) we want (e.g. similar to how we do `DerivedAssets`).  Unfortunately, I couldn't come up with a very clean way of doing this, not without a large restructuring of the project and all submodules with `strings` files, and not without those languages appearing "missing" to `git`.  

3. I was hoping to just have a script modify the project file to only point at the languages we want.  Again, however, this seemed very error prone.

TBH this little hack seems to be the cleanest, for now (and easiest to "undo").

File: Loop/main.swift [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import UIKit
+
+#if FORCE_ENGLISH
+UserDefaults.standard.set(["en"], forKey: "AppleLanguages")
+#else
+UserDefaults.standard.set(nil, forKey: "AppleLanguages")
+#endif
+UserDefaults.standard.synchronize()
+
+UIApplicationMain(
```
Comment (ps2) [inline Loop/main.swift:18]: What does this part of the change do? (changing from synthesized `@UIApplicationMain` to explicit one).

**Review by ps2 — APPROVED** [review]
LGTM; I'm not sure if this would trigger any App Store submission policies (does this override localizations for iOS-built-in text that might be presented from our app, like authorizations and such?); if it does trigger something, we can fall back to removing the localizations during build I suppose.

File: Loop/main.swift (hunk 1 above)
Comment (rickpasetto) [reply Loop/main.swift:18]: `@UIApplicationMain` synthesizes the equivalent of this `main.swift` file.  You have to override the locale even before `UIApplicationMain` is called, which is why I had to create a manual `main.swift`.  As far as I know, there's no way to have any code execute before `UIApplicationMain` if you use `@UIApplicationMain`.

**Comment by rickpasetto** [comment]
@ps2 Good point regarding iOS-built-in text.  The answer, as far as I know, is "no", there's no way to override that.  For example, (in the iOS simulator) if I set my preferred language to Spanish, then, even with this change, an authentication challenge still looks like:

Simulator Screen Shot - iPhone 11 Pro Max - 2021-07-28 at 13 11 20

**Pull request merged**